imports/client/components/CallSection.tsx

```diff
@@ -1,137 +1,159 @@
-      selfProfile,
-      participants,
-      signalsReady,
-      selfUserId,
-      spectraDisabled,
+  useEffect(() => {
+    const observer = ProducerServers.find({ trackId: dupedTrack?.id }).observeChanges({
+      added: (_id, fields) => {
+        producerServerCallback.current?.({ id: fields.producerId! });
+        producerServerCallback.current = undefined;
+      },
+    });
+    return () => observer.stop();
+  }, [dupedTrack]);
+
+  const onProduce = useCallback((
+    { kind, rtpParameters, appData }: {
+      kind: string,
+      rtpParameters:
+      types.RtpParameters,
+      appData: any,
+    },
+    callback: ProducerCallback,
+  ) => {
+    if (dupedTrack?.id !== appData.trackId) {
+      return;
+    }
+
+    producerServerCallback.current = callback;
+    setProducerParams([kind, JSON.stringify(rtpParameters)]);
+  }, [dupedTrack?.id]);
+  useEffect(() => {
+    transport.on('produce', onProduce);
+    return () => {
+      transport.off('produce', onProduce);
     };
-  }, [huntId, puzzleId, tabId]);
+  }, [transport, onProduce]);
 
-  const nonSelfParticipants = useCallback(() => {
-    return tracker.participants.filter((p) => {
-      return (p.createdBy !== tracker.selfUserId) || (p.tab !== tabId);
-    });
-  }, [tracker.participants, tracker.selfUserId, tabId]);
+  useEffect(() => {
+    if (!dupedTrack) {
+      return;
+    }
+    (async () => {
+      const newProducer = await transport.produce({
+        track: dupedTrack,
+        zeroRtpOnPause: true,
+        appData: { trackId: dupedTrack.id },
+      });
+      setProducer(newProducer);
+    })();
+  }, [transport, dupedTrack]);
 
-  const toggleMuted = useCallback(() => {
-    onToggleMute();
-  }, [onToggleMute]);
+  useSubscribe(producerParams ? 'mediasoup:producer' : undefined, transport.appData._id, dupedTrack?.id, ...(producerParams ?? []));
 
-  const toggleDeafened = useCallback(() => {
-    onToggleDeafen();
-  }, [onToggleDeafen]);
+  return <></>;
+};
 
-  const leaveCall = useCallback(() => {
-    onLeaveCall();
-  }, [onLeaveCall]);
+const ProducerBox = ({
+  muted,
+  deafened,
+  audioContext,
+  stream,
+  transport,
+}: {
+  muted: boolean,
+  deafened: boolean,
+  audioContext: AudioContext,
+  stream: MediaStream,
+  transport: types.Transport,
+}) => {
+  const spectraDisabled = useTracker(() => Flags.active('disable.spectra'));
+  const { initial, discordAvatarUrl } = useTracker(() => {
+    const profile = Profiles.findOne(Meteor.userId()!);
+    return {
+      initial: profile ? profile.displayName.slice(0, 1) : 'U', // get it?  it's you
+      discordAvatarUrl: getAvatarCdnUrl(profile?.discordAccount),
+    };
+  });
 
-  const spectrumRefCallback = useCallback((spectrum) => {
-    if (spectrum) {
-      spectrum.connect(localStream);
-    }
-  }, [localStream]);
+  const [tracks, setTracks] = useState<MediaStreamTrack[]>([]);
+  useEffect(() => {
+    const captureTracks = () => Meteor.defer(() => setTracks(stream.getTracks()));
```

Comment:
comment why the eventloop yield is needed (tracks aren't reachable from stream.getTracks() yet)

Reply:
Fixed in 0cd500b